lib/spack/spack/util/web.py

```diff
@@ -508,17 +508,20 @@ def _urlopen(req, *args, **kwargs):
     except AttributeError:
         pass
 
-    # We don't pass 'context' parameter because it was only introduced starting
-    # with versions 2.7.9 and 3.4.3 of Python.
-    if 'context' in kwargs:
-        del kwargs['context']
```

Comment:
Instead of deleting this code, could you do this?  I think it's clearer to get the args right than to catch the exception here:

```python
py_ver = sys.version_info
if py_ver < (2, 7, 9) or ((3, 0) <= py_ver < (3, 4, 3)):
    del kwargs["context"]
```